src/core/option.hpp

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <exception>
+#include <string>
+#include <array>
+
```

Comment:
This class can be used to optionally return objects from functions. I can explain the implementation, but understanding usage of the interface is more important here.

For example:
```c++
class my_class {
    std::vector<int> m_vector;
};

core::option<my_class> maybe_return_object() {
    int error = rand() % 2;
    if (!error) {
        my_class mc;
        return rm; // Return an object, or...
    }
    return {}; // Optionally return nothing.
}

int main() {
    auto ret = maybe_return_object();
    if (ret) { // Cast operator to bool in core::option is overloaded
        // Returned an object! Do something with it...
    } else {
        // Returned an empty object, handle error...
    }
}
```